src/components/App.js

```diff
@@ -1,25 +1,32 @@
-import React from 'react';
+import React, { useState } from 'react';
 import { Switch } from 'react-router-dom';
 import { Route } from 'react-router-dom';
 
 import Menu from './menu/Menu';
 import routes from './routes';
 import ProtectedRoute from './privateRoute/PrivateRoute';
+import Modal from './common/Modal';
+import { ModalProvider } from './common/Modal/ModalContext';
 
 function App() {
+  const [modal, setModal] = useState({ visible: false, component: <></> });
+
   return (
-    <div className="App">
-      <Menu />
-      <Switch>
-        {routes.map((route, index) =>
-          route.private ? (
-            <ProtectedRoute key={index} {...route} />
-          ) : (
-            <Route key={index} {...route} />
-          )
-        )}
-      </Switch>
-    </div>
+    <ModalProvider value={{ modal, setModal }}>
```

Comment:
I think you could avoid passing the modal value in the context and using it directly in the Modal component as a prop.